WriteVHDLSyntax.py

```diff
@@ -934,52 +983,59 @@ def writeTBMemoryWriteRAMInstance(mtypeB, proc, bxbitwidth, is_binned):
     # FIX ME change number of bins from default 8 to 16 for VMSME Disk memories
     string_mem += "  -- FIX ME change number of bins from default 8 to 16 for VMSME Disk memories!!!\n" if "VMSME" in mtypeB else ""
 
-    string_mem += "  "+mtypeB+"_loop : for var in enum_"+mtypeB+" generate\n"
-    string_mem += "  begin\n"
-    string_mem += "    write"+mtypeB+" : entity work.FileWriterFromRAM" + ("Binned\n" if is_binned else "\n")
-    string_mem += "    generic map (\n"
-    string_mem += "      FILE_NAME".ljust(str_len)+"=> FILE_OUT_"+mtypeB+"&memory_enum_to_string(var)&outputFileNameEnding,\n"
-    string_mem += "      RAM_WIDTH".ljust(str_len)+"=> " + mtypeB.split("_")[1] + ",\n"
-    string_mem += "      NUM_PAGES".ljust(str_len)+"=> " + str(2**bxbitwidth) + "\n"
-    string_mem += "    )\n"
-    string_mem += "    port map (\n"
-    string_mem += "      CLK".ljust(str_len)+"=> CLK,\n"
-    string_mem += "      ADDR".ljust(str_len)+"=> "+mtypeB+"_mem_AV_readaddr(var),\n"
-    string_mem += "      DATA".ljust(str_len)+"=> "+mtypeB+"_mem_AV_dout(var),\n"
-    string_mem += "      READ_EN".ljust(str_len)+"=> "+mtypeB+"_mem_A_enb(var),\n"
-    string_mem += "      NENT_ARR".ljust(str_len)+"=> "+mtypeB+"_mem_AA" + ("A" if is_binned else "") + "V_dout_nent(var),\n"
-    string_mem += "      DONE".ljust(str_len)+"=> "+proc+"_DONE\n"
-    string_mem += "    );\n"
-    string_mem += "  end generate "+mtypeB+"_loop;\n\n"
+    memList = memDict[mtypeB]
+
+    for memMod in memList:
+
+        mem = memMod.list
```

Comment:
@aryd The `MemModule` class doesn't have a member called `list`. Did you mean `inst`?

```suggestion
        mem = memMod.inst
```